binder/src/main/java/io/grpc/binder/internal/ActiveTransportTracker.java

```diff
@@ -0,0 +1,114 @@
+package io.grpc.binder.internal;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import java.util.HashSet;
+import javax.annotation.concurrent.GuardedBy;
+import io.grpc.Attributes;
+import io.grpc.Metadata;
+import io.grpc.internal.ServerStream;
+import io.grpc.internal.ServerTransportListener;
+
+/**
+ * Tracks which {@link BinderTransport.BinderServerTransport} are currently active and allows
+ * invoking a {@link BinderTransportSecurity.TerminationListener} only once all transports are
+ * terminated.
+ */
+final class ActiveTransportTracker {
+  @GuardedBy("this")
+  private final HashSet<BinderTransport.BinderServerTransport> activeTransports = new HashSet<>();
+  private final BinderTransportSecurity.TerminationListener transportSecurityTerminationListener;
```

Comment:
There's nothing security specific about this class (which is good) so can you make this just a `Runnable` and call it `terminationListener` ?

Reply:
I originally did it this way because I thought it might be a good idea to hint why the termination listener was needed, since it wasn't used for anything else. I don't feel super strongly about it though. Updated.